app/main/viewmanager.ts

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+import { BrowserWindow, ipcMain } from 'electron';
+import { View, ViewProps } from './view';
+
+import ConfigUtil = require('../renderer/js/utils/config-util');
+
+class ViewManager {
+	views: { [key: number]: View };
+	selectedIndex: number;
+
+	constructor() {
+		this.views = {};
+		this.selectedIndex = 0;
+		this.registerIpcs();
+	}
+
+	registerIpcs(): void {
+		ipcMain.on('view-create', (e: Event, props: ViewProps) => {
+			this.create(props);
+		});
+
+		ipcMain.on('view-select', (e: Event, index: number) => {
+			this.select(index);
+		});
+
+		ipcMain.on('view-destroy', (e: Event, index: number) => {
+			this.destroy(index);
+		});
+
+		ipcMain.on('view-destroy-all', () => {
+			this.destroyAll();
+		});
+
+		ipcMain.on('forward-message-view', (e: Event, name: string, ...params: any[]) => {
+			this.views[this.selectedIndex].webContents.send(name, ...params);
+		});
+
+		ipcMain.on('forward-message-all', (e: Event, name: string, ...params: any[]) => {
+			this.forwardMessageAll(name, ...params);
+		});
+
+		ipcMain.on('call-view-function', (e: Event, name: string, ...params: any[]) => {
+			const fn = this.views[this.selectedIndex][name as keyof View];
+			if (typeof fn === 'function') {
+				// Type checking requires spread elements to match up with a rest parameter.
+				// So, using a workaround here.
+				(fn as any)(...params);
+			}
+		});
+	}
+
+	create(props: ViewProps): void {
+		if (this.views[props.index]) {
```

Comment:
I think I mentioned this before but, the allowing a programmer to pass and track IDs to pass into this `index` field here is very brittle; it seems to call for trouble. Can we make this auto generated and make `readonly` property?